add Diplom 3 and allure

VladSem95 opened the pull request:
Работа выполнена в соответсвии с заданием(см. ниже). Использовалась сторонняя библиотека Faker для генерации тестовых данных. Для поиска элементов использовался xpath. Для Яндекс браузера в предусловиях тестов закомментировано по 2 строки. Для прогона тестов в Яндекс браузере необходимо просто раскомментировать эти 2 строки. Кроме этого испльзовал Rest Assured для создание тестовых пользователей и для удаления их после теста.
На случай, когда регистрация не проходит добавил обработчик исключения try - catch для NullPointerException.


Регистрация
Проверь:
Успешную регистрацию.
Ошибку для некорректного пароля. Минимальный пароль — шесть символов.
Вход 
Проверь:
вход по кнопке «Войти в аккаунт» на главной,
вход через кнопку «Личный кабинет»,
вход через кнопку в форме регистрации,
вход через кнопку в форме восстановления пароля.
Переход в личный кабинет 
Проверь переход по клику на «Личный кабинет».
Переход из личного кабинета в конструктор 
Проверь переход по клику на «Конструктор» и на логотип Stellar Burgers.
Выход из аккаунта
Проверь выход по кнопке «Выйти» в личном кабинете.
Раздел «Конструктор»
Проверь, что работают переходы к разделам:
«Булки»,
«Соусы»,
«Начинки».

VladSem95 committed:
add Diplom 3 and allure

mkouznetsov-git commented on src/test/java/JumpToTabParamTest.java:
```diff
@@ -0,0 +1,72 @@
+        //клик по соседнему элементу, чтобы усмотреть кликабельность секции "Булки" при открытии стартовой страницы
+        driver.findElement(neighboringSection).click();
+        new WebDriverWait(driver, Duration.ofSeconds(3));
+        driver.findElement(sectionConstructorTab).click();
```
Если удалить эту строчку, то тест всё-равно успешно проходит, это значит, что на самом деле он не проверяет переход по вкладкам.

VladSem95 replied:
Ресерчил без этой строки на момент падения теста. При открытии стартовой страницы у нас по умолчанию активна таба "Булки" и поэтому она является некликабельной. Эта строка не мешает тестам, но дает возможность этот тест включить в параметризированный.

mkouznetsov-git commented on src/test/java/LoginUserParamTest.java:
```diff
@@ -0,0 +1,86 @@
+        driver.findElement(By.xpath(buttonXpath)).click();
+        loginPage.inputLoginDataAndPressButton(driver,userEmail,userPassword);
+        new WebDriverWait(driver, Duration.ofSeconds(3))
+                .until(ExpectedConditions.elementToBeClickable(homePage.getXpathCheckoutButtonText(driver)));
```
homePage.getXpathCheckoutButtonText(driver) -- для доступа к статичным методам надо использовать имя класса, а не объекта. Проверь по всему проекту -- есть ещё места с этой ошибкой.

mkouznetsov-git commented on src/test/java/RegistrationTest.java:
```diff
@@ -0,0 +1,87 @@
+        registrationPage.inputRegistrationDataAndPressButton(driver, userName, userEmail, userPassword);
+        loginPage.waitLoadInputButton();
+        MatcherAssert.assertThat(loginPage.getLoginTextButton(driver),containsString("Войти"));
+        driver.quit();
```
Если тест упадёт где-то выше, то до этой строчки дело не дойдёт и браузер останется открытым.

VladSem95 committed:
add change and new allure

mkouznetsov-git commented on src/test/java/RegistrationTest.java:
```diff
@@ -0,0 +1,83 @@
+                String clientBearerToken = login.extract().path("accessToken");
+                clientBearerToken = clientBearerToken.replace("Bearer ", "");
+                userClient.delete(clientBearerToken);
+                driver.quit();
```
Две одинаковые строки и в блоке try и в блоке catch. Вынеси эту команду до блока try..catch, а две лишние убери.

mkouznetsov-git commented on src/test/java/RegistrationTest.java:
```diff
@@ -0,0 +1,83 @@
+                clientBearerToken = clientBearerToken.replace("Bearer ", "");
+                userClient.delete(clientBearerToken);
+                driver.quit();
+            }catch(NullPointerException Exception){
```
Переменная должна называться с маленькой буквы.
Чтобы проигнорировать блок catch() обычно пишут так:
`catch(NullPointerException ignore){}`

mkouznetsov-git commented on src/test/java/JumpToTabParamTest.java:
```diff
@@ -0,0 +1,71 @@
+        driver.manage().window().maximize();
+        driver.get(HomePage.openHomePage());
+        //клик по соседнему элементу, чтобы учесть НЕкликабельность секции "Булки" при открытии стартовой страницы
+        driver.findElement(neighboringSection).click();
```
Если удалить эту и следующие две строчки, то тест проходит успешно. Значит, проверка именно переходов по табам не реализована. Надо поправить.

VladSem95 committed:
add change#2:
 1. RegistrationTest: поправил блок try, catch.
 2. RegistrationTest: изменил регистр переменной класса NullPointerException + изменил названил на ignore
 3.JumpToTabParamTest: изменил тест, в текущей реализации происходит проверка на то, что таб стал активным.
 4.HomePage: добавил локаторы для активных табов + статические методы, которые возвращают локаторы.

 new allure
